packages/db/src/translate/executions.ts

```diff
@@ -0,0 +1,71 @@
+import Knex from 'knex';
+
+import { ExecutionRecord } from '@cumulus/types/api/executions';
+import { PostgresExecution } from '../types';
+// TODO move to common location
+const {
+  getParentExecutionCumulusId,
+  getAsyncOperationCumulusId,
+  getCollectionCumulusId,
+} = require('../../api/lambdas/sf-event-sqs-to-db-records/utils');
+
+/**
+ * Translate execution record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Source record from DynamoDB
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} knex
+ *   Knex client
+ * @returns {PostgresExecutionRecord} - converted Execution
+ */
+export const translateApiExecutionToPostgresExecution = async (
+  dynamoRecord: ExecutionRecord,
+  knex: Knex
+): Promise<PostgresExecution> => {
+  // Map old record to new schema.
+  const translatedRecord: PostgresExecution = {
+    async_operation_cumulus_id: (
+      dynamoRecord.asyncOperationId ? await getAsyncOperationCumulusId(
+        dynamoRecord.asyncOperationId, knex
+      ) : undefined
+    ),
+    status: dynamoRecord.status,
+    tasks: JSON.stringify(dynamoRecord.tasks),
+    error: JSON.stringify(dynamoRecord.error),
+    arn: dynamoRecord.arn,
+    duration: dynamoRecord.duration,
+    original_payload: JSON.stringify(dynamoRecord.originalPayload),
+    final_payload: JSON.stringify(dynamoRecord.finalPayload),
+    workflow_name: dynamoRecord.name,
+  };
+
+  if (dynamoRecord.timestamp) {
+    translatedRecord.timestamp = new Date(Number(dynamoRecord.timestamp));
+  }
+
+  if (dynamoRecord.collectionId) {
+    // TODO is there a helper for this?
+    const collectionNameVersionArray = dynamoRecord.collectionId.split('___');
```

Comment:
I didn't see one that goes from the `collection___version` string, back to an array or back to a cumulus_id. Again, this will likely end up on the new PG model for Collections I'd think. Looking at it now as part of the merge of the updated rds branch.

Reply:
On second thought, we shouldn't need this after the migration. I'm just going to leave it as-is rather than put something on the PG model that's Dynamo-specific or further extend the deprecated Dynamo model.

Reply:
I just found this `deconstructCollectionId` which is what you may have been looking for:

https://github.com/nasa/cumulus/blob/master/packages/api/lib/utils.js#L34-L40

```
function deconstructCollectionId(collectionId) {
  const [name, version] = collectionId.split('___');
  return {
    name,
    version,
  };
}
```

Reply:
Yeah, this is exactly what I was looking for. Thanks for finding it, sorry to miss it!

I guess I'm a little hesitant to use it though. Looking at the @cumulus/db package where my code in question lives, we're not currently importing @cumulus/api (where this util lives). I'm a little reluctant to pull in the whole api package  for just this string parsing so I think that might mean moving the `deconstructCollectionId` util to @cumulus/common or something like that. At that point is it worth it for one line that is only used for decoding the deprecated Dynamo field? I guess I'm leaning towards leaving the one line as-is even though it's not very DRY. I also don't feel super strongly so feel free to disagree 😄

Reply:
I wouldn't want to move it to `common` because we're trying to keep that package really slim, since it used to be a dumping ground for everything. But we could move it somewhere like this:

https://github.com/nasa/cumulus/blob/feature-rds/packages/message/src/Collections.ts

You'd still be bringing in that whole package just to use that helper though, which is unfortunate. So I could go either way

Reply:
This is kind of thinking ahead... but as I'm implementing my work for the granules and files migration, I will need to deconstruct the `collectionId` of a granule to refer to a PG collection record and retrieve the `collection_cumulus_id`. So, we will at least know `deconstructCollectionId` will be used in one other place if you do decide to bring it in 🧐